src/ports/postgres/modules/graph/pagerank.py_in

```diff
@@ -527,14 +562,63 @@ def pagerank(schema_madlib, vertex_table, vertex_id, edge_table, edge_args,
                 """.format(**locals()))
 
         # Step 4: Cleanup
-        plpy.execute("""DROP TABLE IF EXISTS {0},{1},{2},{3},{4},{5},{6}
+        plpy.execute("""DROP TABLE IF EXISTS {0},{1},{2},{3},{4},{5},{6},{7}
             """.format(out_cnts, edge_temp_table, cur, message, cur_unconv,
-                       message_unconv, nodes_with_no_incoming_edges))
+                       message_unconv, nodes_with_no_incoming_edges, personalized_nodes))
         if grouping_cols:
             plpy.execute("""DROP TABLE IF EXISTS {0},{1},{2}
                 """.format(vertices_per_group, temp_summary_table,
                            distinct_grp_table))
 
+
+def get_query_params_for_ppr(nodes_of_interest, damping_factor,
+                             ppr_join_clause, vertex_id, edge_temp_table, vertex_table,
+                             cur_distribution, personalized_nodes):
+    """
+     This function will prepare the Join Clause and the condition to Calculate the Personalized Page Rank
+     and Returns Total number of user provided nodes of interest, A join Clause and a clause to be added
+     to existing formula to calculate pagerank.
+
+     Args:
+         @param nodes_of_interest
+         @param damping_factor
+         @param ppr_join_clause
+         @param vertex_id
+         @param edge_temp_table
+         @param vertex_table
+         @param cur_distribution
+
+     Returns :
+             (Integer, String, String)
+
+    """
+    total_ppr_nodes = 0
+    random_jump_prob_ppr = ''
```

Comment:
This is more like a sql condition/clause, not a number of prob, should we rename it somehow?

Reply:
renamed this variable to ppr_random_prob_clause